api/datasets_test.go

```diff
@@ -0,0 +1,155 @@
+package api
+
+import (
+	"net/http"
+	"strconv"
+	"testing"
+)
+
+func TestMakeGetBodyParams(t *testing.T) {
+	cases := []struct {
+		description                   string
+		download                      bool
+		format                        string
+		page, pageSize, offset, limit int
+		expOffset, expLimit           int
+		all, readOnly                 bool
+		expAll                        bool
+		expectedErr                   string
```

Comment:
nice! lately I've been splitting table tests into two slices, one for the "good" (cases that shouldn't error), and another for the "bad" (cases that should _only_ produce errors).  Example: https://github.com/qri-io/qri/blob/master/lib/datasets_test.go#L82-L136

mind splitting these? I think you'll like the result. In the meantime I'll document this over on that code conventions issue...

Reply:
yea that is awesome, way less mental gymnastics.